backend/src/services/file.ts

```diff
@@ -38,37 +35,11 @@ export async function uploadFile(user: UserInterface, modelId: string, name: str
 
   await file.save()
 
-  if (config.avScanning.enabled) {
-    if (!av) {
-      try {
-        av = await new NodeClam().init({ clamdscan: config.avScanning.clamdscan })
-      } catch (error) {
-        log.error(error, 'Unable to connect to ClamAV.')
-        return file
-      }
-    }
-    const avStream = av.passthrough()
-    const s3Stream = (await getObjectStream(file.bucket, file.path)).Body as Readable
-    s3Stream.pipe(avStream)
-    log.info({ modelId, fileId: file._id, name }, 'Scan started.')
-
-    avStream.on('scan-complete', async (result) => {
-      log.info({ result, modelId, fileId: file._id, name }, 'Scan complete.')
-      await file.update({
-        avScan: { state: ScanState.Complete, isInfected: result.isInfected, viruses: result.viruses },
-      })
-    })
-    avStream.on('error', async (error) => {
-      log.error({ error, modelId, fileId: file._id, name }, 'Scan errored.')
-      await file.update({
-        avScan: { state: ScanState.Error },
-      })
-    })
-    avStream.on('timeout', async (error) => {
-      log.error({ error, modelId, fileId: file._id, name }, 'Scan timed out.')
-      await file.update({
-        avScan: { state: ScanState.Error },
-      })
+  if (config.ui.avScanning.enabled) {
+    const fileScanConnectors = getFileScanningConnectors()
```

Comment:
It would be good if this new connector type could have the same interface as other connectors and this could just be `scanners.scan(file)`. Ideally we want to encapsulate the initialisation and the calling of each individual scanner inside the connector logic, not here in the service layer every time we want to use the scanners